<launch>
  <!--Launch the PID controllers for velocity and steering-->
  <!--The PID is from the ROS pid package-->
  <!--Velocity Controlller-->

  <node name="pid_controller" pkg="pid" type="controller" ns="velocity">
    <param name="Kp" value="5.0"/>
    <param name="Ki" value="0.0"/>
    <param name="Kd" value="0.0"/>
    <param name="upper_limit" value="200"/>
    <param name="lower_limit" value="-200"/>
    <param name="windup_limit" value="10"/>
    <param name="max_loop_frequency" value="100.0"/>
    <param name="min_loop_frequency" value="100.0"/>
  </node>

  <node name="pid_controller" pkg="pid" type="controller" ns="steering">
    <param name="Kp" value="5.0"/>
    <param name="Ki" value="0.0"/>
    <param name="Kd" value="0.0"/>
    <param name="upper_limit" value="100"/>
    <param name="lower_limit" value="-100"/>
    <param name="windup_limit" value="100"/>
    <param name="max_loop_frequency" value="100.0"/>
    <param name="min_loop_frequency" value="100.0"/>
    <param name="angle_error" value="true"/>
    <param name="angle_wrap" value="$(eval 2.0*180.0)"/>
  </node>

  <!--Reconfig allows for retunning of the PIDS-->
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>
  <node name="rqt_plot" pkg="rqt_plot" type="rqt_plot"/>
  <!--Movement_Controller-->
  <node name="movement_controller" pkg="smile_mobile_robot" type="movement_controller.py"/>


</launch>
